radis/lbl/calc.py

```diff
@@ -66,6 +66,7 @@ def calc_spectrum(
     export_lines=False,
     verbose=True,
     return_factory=False,
+    engine="pandas",
```

Comment:
Adding a new parameter makes the library more difficult to understand & use for new users; ideally we'd like all the input parameters of calc_spectrum() to be only physical parameters (pressure, temperature, species, etc.)  but not computational parameters. 

One alternative I'm trying to push is to store the "engine" key in `radis.config` , which itself is read from the radis.json file. Each user can modify their own radis.json file. The default one is copied from [radis/default_radis.json](https://github.com/radis/radis/blob/develop/radis/default_radis.json) . All of this is implemented and working quite well. 

Have a look at how `MEMORY_MAPPING_ENGINE` is implemented there and in the code. You could do a similar thing for a new key `DATAFRAME_ENGINE`, for instance (default "pandas" then "auto" or "vaex" if it works well !) 

Users can edit it either in radis.json ; or in their scripts : 
```python
import radis
radis.config["DATAFRAME_ENGINE"] = "vaex"
```


Also tagging @minouHub for reference

Reply:
I think as the default value of engine is pandas, everything thing should work fine for new users and all the examples given in docs can be tested without bothering about engine parameter of calc_spectrum().Just by supplying physical conditions , user can calculate spectrum easily.
Also, updating the radis.json every time user wants to switch from calculating spectrum with a low number of lines to spectrum having a high number of lines is not that flexible.

Reply:
If you want changes i can make those.

Reply:
Our target users aren't spectroscopy experts; and shouldn't even have to worry about which mode to use. Less parameters = less trouble for them, so I'd like not to have a new parameter unless there is a strong argument against it. There are already too many parameters in calc_spectrum (and we'll have to remove some)  [just have a look at [calc_spectrum](https://radis.readthedocs.io/en/latest/source/radis.lbl.calc.html#radis.lbl.calc.calc_spectrum) docs, it's frightening for a 1st time user ! :) )  

What you're suggesting (switch from one to another) is interesting and could (should !) be done under-the-hood, without the user noticing : 
- there could be an "auto" mode that automatically switch if we evaluate that Vaex might be faster when loading the database, 

Do not implement this now, as the goal is to merge the Vaex calculations (it won't be the default mode yet); but keep in mind that this is where we're heading to so that the underlying architecture is ready for this "auto" mode. 

So to wrap is up

1. for the time being, use `DATAFRAME_ENGINE = "pandas"` as default, but `DATAFRAME_ENGINE = "vaex"` should work 
2. we'll publish a new Radis release and ask people in the Slack to use DATAFRAME_ENGINE = "vaex" and see if it works for them, gather some user feedbacks
3. if it works well we can implement `DATAFRAME_ENGINE = "auto"`

Reply:
I will make the changes.